src/reference/asciidoc/kafka.adoc

```diff
@@ -1171,6 +1171,22 @@ public KafkaTemplate<String, String> myReplyingTemplate() {
 }
 ----
 
+IMPORTANT: If the listener method returns `Message<?>` or `Collection<Messge<?>>` the listener method is responsible for setting up the message headers for the reply; for example, when handling a request from a `ReplyingKafkaTemplate`, you might do the following:
+
+[source, java]
+----
+@KafkaListener(id = "messageReturned", topics = "someTopic")
+public Message<?> listen(String in, @Header(KafkaHeaders.REPLY_TOPIC) byte[] replyTo,
```

Comment:
Are you missing `@SendTo` here?
Or it doesn't matter and non-`void` return is enough to produce reply?

Reply:
Interesting; yes; I guess if the return type is `Message<?>` the `@SendTo` is entirely optional - we could enforce it, I suppose, but maybe it's ok. 

WDYT?

Reply:
Ah! Yes, it's really OK: your sentence really implies that.
So, yeah, merging...
Thanks